educator/src/Dscp/Educator/Web/Auth.hs

```diff
@@ -69,21 +74,47 @@ instance ( HasServer api ctxs
                 (Authenticated (r :: res)) -> return r
                 _ -> delayedFailFatal $ err401 { errHeaders = [("WWW-Authenticate", authMsg)] }
 
-instance FromJSON AuthData
-instance FromJWT AuthData
-
 ---------------------------------------------------------------------------
 -- Helpers
 ---------------------------------------------------------------------------
 
+-- | If request has header "Authorization: Bearer <token>", get the
+-- "<token>" part.
+authBearerToken :: Request -> Maybe ByteString
+authBearerToken =
+    L.lookup "Authorization" . requestHeaders >=>
+    BS.stripPrefix "Bearer "
+
+checkJWitness :: AuthCheck (PublicKey, ByteString)
+checkJWitness = do
+    request <- ask
+    token <- maybe mempty pure $ authBearerToken request
+    either (const mempty) pure $ do
+        jWitness <- first (show @Text @Error) $
+            decodeCompact $ LBS.fromStrict token
+        verifyJWitness jWitness
+
+-- | Authentication header timeout in seconds.
+-- TODO: 1) make it configurable via config file
+--       2) isn't 5 minutes too long?
```

Comment:
It was set to 5 minutes to account for the fact that the client might have their system clock behind by a couple of minutes.